### PR TITLE
Fixes #51 - dependency management clean-up

### LINKED ISSUE #51: Use latest WildFly BOM instead of the old EAP one
It could help to resolve some dependency issues.

This change is related also to @mchoma findings in https://issues.jboss.org/browse/JBEAP-5654

### PR DESCRIPTION
**Pull request by kwart**
Fixes #51

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,9 @@
 - added new configuration entries to Bare Metal provider:
   - `baremetal.ssh.port` - port number of SSH server in the Node
   - `baremetal.privateAddress` - configurable value for `getPrivateAddress()` method
+- WildFly 10.1.0 BOM replaces JBoss EAP 6.4 BOM in dependencyManagement
```
Comment (Ladicek) [inline CHANGELOG.md]: Probably want to put "dependencyManagement" into backticks.

File: CHANGELOG.md [hunk 2]
```diff
@@ -20,6 +20,9 @@
...
+- removed transitive dependency on JBoss Controller client; users have to define it manually
+  in the same way as when using standalone Creaper library
```
Comment (Ladicek) [inline CHANGELOG.md]: Should be mentioned somewhere in the README or some other prominent place.

**Comment by Ladicek** [comment]
Otherwise LGTM.

File: CHANGELOG.md (hunk 2 above)
Comment (kwart) [inline CHANGELOG.md]: Agree, probably the `wildfly-README.md` and `arquillian-README.md` are good candidates.

**Comment by Ladicek** [comment]
Yea, looks good.

**Comment by kwart** [comment]
OK, I'll squash them.

**Comment by kwart** [comment]
Merging. Thanks for review.

**Pull request merged**